contract_payment_mode/models/contract.py

```diff
@@ -13,7 +13,14 @@ class ContractContract(models.Model):
 
     @api.onchange("partner_id")
     def on_change_partner_id(self):
-        self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+        if self.contract_type == "purchase":
+            self.payment_mode_id = self.with_company(
+                self.company_id
+            ).partner_id.supplier_payment_mode_id.id
+        else:
+            self.payment_mode_id = self.with_company(
+                self.company_id
+            ).partner_id.customer_payment_mode_id.id
```

Comment:
More readable, but a nitpicking:
```suggestion
        partner = self.with_company(self.company_id).partner_id
        if self.contract_type == "purchase":
            self.payment_mode_id = partner.supplier_payment_mode_id.id
        else:
            self.payment_mode_id = partner.customer_payment_mode_id.id
```

Reply:
@pedrobaeza perfect.. I'll adjust.
Thanks

Reply:
@pedrobaeza Done.